presto-native-execution/presto_cpp/main/runtime-metrics/RuntimeMetricsReporter.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/runtime-metrics/RuntimeMetricsReporter.h"
+#include "presto_cpp/main/common/Configs.h"
+#include "presto_cpp/main/runtime-metrics/prometheus-reporter/PrometheusReporter.h"
+
+namespace facebook::presto {
+namespace {
+
+std::string getSerializedMetrics() {
+#ifdef PRESTO_ENABLE_PROMETHEUS_REPORTER
+  if (auto reporter = std::dynamic_pointer_cast<prometheus::PrometheusReporter>(
+          folly::Singleton<velox::BaseStatsReporter>::try_get())) {
+    return reporter->getSerializedMetrics();
+  }
+#endif
+  /// This method shouldn't invoked if Prometheus is not enabled.
+  VELOX_UNREACHABLE("No metrics reporter is enabled");
```

Comment:
```suggestion
  auto reporter = std::dynamic_pointer_cast<prometheus::PrometheusReporter>(
          folly::Singleton<velox::BaseStatsReporter>::try_get()) 
  VELOX_CHECK_NOT_NULL(reporter, "Stats reporter singleton is not registered");
  return reporter->getSerializedMetrics();
```

Reply:
It makes sense to add that interface. It should be a common scenario for reporters (although meta internal does not need this method, we can just do noop)